custom_components/chargeamps/sensor.py

```diff
@@ -106,3 +115,57 @@ def device_state_attributes(self):
     def unique_id(self):
         """Return a unique ID to use for this sensor."""
         return f"{DOMAIN}_{self.charge_point_id}_{self.connector_id}"
+
+
+class ChargeampsPowerSensor(ChargeampsSensor):

```

Comment:
I thought it would be convenient to reuse the `ChargeampsSensor` class, but as `ChargeampsPowerSensor` defines a `device_class` are we not allowed to set `icon`. As this just is a `custom_component` it wouldn't matter but I might be good to know.

https://github.com/kirei/hass-chargeamps/pull/5/files#diff-a3c3ebd72271874b2a39bfc4089033cdR100

Reply:
PYou can create a base class based on  ChargeampsSensor and use that for both ChargeampsSensor ChargeampsPowerSensor?

Reply:
See https://github.com/kirei/hass-chargeamps/tree/base_entity

Reply:
True.
And then I could use that class in #4 too. The base class should not have the `icon` method defined at all though. 

Let me know and I'll rebase against that.

Reply:
Please check https://github.com/kirei/hass-chargeamps/pull/6